### PR TITLE
[FLINK-33379] Bump CI flink version on flink-connector-elasticsearch

### PR DESCRIPTION
**Pull request by liyubin117**
As Flink 1.18 released, bump the flink version in elasticsearch connector with push_pr.yml and weekly.yml

### CODE REVIEW

**Comment by liyubin117** [comment]
@MartijnVisser I have updated CI yml files and pined commons-compress and snappy-java version for guaranteeing dependency convergence, PTAL, Thanks!

File: .github/workflows/weekly.yml [hunk 1]
```diff
@@ -35,12 +35,18 @@ jobs:
...
+          branch: main
         }, {
           flink: 1.16.2,
           branch: v3.0
         }, {
           flink: 1.17.1,
           branch: v3.0
+        }, {
+          flink: 1.18.0,
+          branch: v3.0
```
Comment (MartijnVisser) [inline .github/workflows/weekly.yml:49]: Does the v3.0 branch support 1.18.0?

Comment (liyubin117) [reply .github/workflows/weekly.yml:49]: I found that you removed the 1.18.0 support in the follow hotfix, and there are compilation failures indeed, so I will 
 also remove the support. thanks for your reminds.
image

**Comment by liyubin117** [comment]
@MartijnVisser CI has succeed.
[https://github.com/liyubin117/flink-connector-elasticsearch/actions/runs/6692027990](url)

**Comment by pgodowski** [comment]
What is the expected timeline for having the flink-connector-elasticsearch7 [published](https://central.sonatype.com/artifact/org.apache.flink/flink-connector-elasticsearch7/versions), with the Flink 1.18 support?

**Pull request merged**